### PR TITLE
Prevent state updates on unmounted components

### PR DESCRIPTION
**Pull request by sherryhli**
Discovered some console warnings caused by state updates on unmounted components. In the future we need to check if the component has unmounted when doing state updates in async callbacks.

console_warning

### CODE REVIEW

File: src/Components/Application/Application.jsx [hunk 1]
```diff
@@ -62,32 +62,49 @@ function Application({ applications, newReview, history, match, user }) {
...
+
     if (appId == null || user == null || review == null) {
       return;
     }
     UPDATE.updateReviewAPI(review).then((res) => {
       if (!isRated.current && review.rating > -1) {
         isRated.current = true;
-        newReview();
+        if (!hasUnmounted) {
+          newReview();
```
Comment (xinhaoz) [inline src/Components/Application/Application.jsx:97]: I think it's fine to leave this one, as newReview is actually a dispatcher for the global redux state

**Comment by GodGreg** [comment]
> After looking at this, I think a really fun/neat task would be to create a custom hook for this pattern, something like in https://www.robinwieruch.de/react-hooks-fetch-data but changed slightly to fit our use cases.
> If you want to take that task RN feel free to do it in this PR but I'd also be fine with having this temporarily (I'll probably do the custom hook in the near future just because it sounds like fun, but if anyone reviewing sees this lmk if you have an interest in writing/pairing on it!)
> lgtm after addressing minor comment

Interesting. I haven't used hooks a ton so I was unaware of state updates on unmounted components. It is something I plan to put more research into so I understand how hooks work at a low level compared to lifecycle methods. 

@xinhaoz I would be interested in doing a pair program on this if I have the bandwidth. Sounds cool.

**Comment by sherryhli** [comment]
@ahmedhamodi Sorry, late response! I can take care of it

**Comment by ahmedhamodi** [comment]
Great! Let me know if there is anything I can help with.

**Comment by sherryhli** [comment]
Closing this, issue addressed in #118

**Pull request closed without merging**